### ISSUE TITLE
The zip method produces a List of plain Arrays instead of List of List

### ISSUE THREAD

**Issue by vcarel** [facebook/immutable-js#1236]
Following the same example as in #1235:

    const Immutable = require('immutable')

    const hits = Immutable.fromJS([{ price: 1 }, { price: 2 }, { price: 3 }, { price: 4 }, { price: 5 }])

    const evens = hits.filter((value, index) => index % 2 === 0)
    const odds = hits.filter((value, index) => index % 2 === 1)
    const pairs = evens.zip(odds)

    console.log('Is the list of pairs a plain old Array ?', Array.isArray(pairs))
    console.log('Is the first pair a plain old Array ?', Array.isArray(pairs.get(0)))

To the above question, the answers are yes (true) and no (false).

In other words, the ```zip``` methods produces a ```List``` of pairs (as expected), but each pair is a plain old ```Array``` object (which is totally unexpected)

**Comment by zbrukas** [comment]
What is your immutable version? Reproducing with 3.8.1 gave no errors.

**Comment by vcarel** [comment]
@zbukras I used v3.8.1. What do you mean by "no errors" ?
The point is that `zip` produces a mix of Immutable Lists and Javascript Arrays...

**Comment by leebyron** [comment]
Thanks for reporting this. Right now it's working as designed, since List is for a collection of same value type, while Arrays are often used as "tuples" of a fixed size and a different value at each index.

I think it's worth discussing this further since there's bound to be tradeoffs to this approach. Perhaps you can share more about how the plain tuple array limits your use of the library? Or is it just about expectations (e.g. should this just result in documentation changes?)

**Comment by leebyron** [comment]
As of #1369 Arrays are fully supported as nested collections, while there were likely some rough trade-offs before, there should be much better support for different nested collection types so I think it's okay to stick with the current behavior.
